### ISSUE TITLE
"Source of truth" should be Nuget

### ISSUE THREAD

**Issue by Jericho** [cake-contrib/Cake.AddinDiscoverer#22]
Status.md and YAML files on Cake's web site are maintained manually and therefore error prone and also it's a tedious task. This should be automated.

I propose the following:

- [x] Remove the "Addins" and "Recipes" from Status.md. We can keep other sections such as "Deprecated" and continue maintaining them manually
- [x] AddinDiscover should query Nuget.org for all package matching "Cake.*". This logic would exclude any addin that does not follow the recommended naming convention. Currently, there is only one addin (MagicChunks) that does not follow this convention, therefore I think this is acceptable.
- [x] AddinDiscoverer should sync the YAML files with the list of packages discovered on Nuget
    - [x] Delete YAML if it doesn't have a corresponding nuget package
    - [x] Create YAML if nuget package doesn't have one already
    - [x] Update existing YAML if necessary
- [x] AddinDiscoverer should submit PR with proposed changes
- [x] AddinDiscoverer should not submit another PR if a previous one is still open

**Comment by gep13** [comment]
This sounds great to me.  If we could keep a note of any that don't match when going through this process, that would be great as well.

**Comment by Jericho** [comment]
- [ ] Generate a separate report for Nuget packages containing recipes
- [ ] Generate a separate report for Nuget packages containing Modules

**Comment by Jericho** [comment]
- [ ] Discontinue reporting whether an addin has a corresponding YAML file on the cake web site since yaml files are now automatically synchronized

**Comment by Jericho** [comment]
@gep13 I was about to submit PR with changes to status.md but it occurred to me that almost 100% of the info in this file is now automatically generated by the audit process. Therefore I'm wondering if we could simply get rid of this file completely.

For instance:
1. The `Recipes` section is superseded by `Audit_for_recipes.md`
2. The `Modules` section is superseded by `Audit_for_Cake_x.y.z.md` which reports on addins and modules
3. The `Websites` section does not have an equivalent generated by the audit tool but there's only a single entry in this section (I don't even know what is the purpose of this section)
4. The `Addins` section is superseded by `Audit_for_Cake_x.y.z.md` which reports on addins and modules
5. The `Work In Progress` section does not really have an equivalent generated by the audit tool but since I modified the logic to audit 'pre-release' packages, I suspect that the vast mojority of the addins that were listed in this section will be automatically listed in `Audit_for_Cake_x.y.z.md`
6. The `Needs investigation` section: see my comment for the `Work In Progress` section
7. The `Deprecated` section is superseded by the `Deprecated` section in `Audit.md`

In conclusion, I think we can delete Status.md and simply rely on the various markdown file generated by the audit process.

**Comment by Jericho** [comment]
@gep13 I submitted a [PR](https://github.com/cake-contrib/Home/pull/54) with changes to Status.md but let me know what you think about deleting completely the file.

**Comment by Jericho** [comment]
Version 3.1.0 published

**Comment by gep13** [comment]
@Jericho for now, I think I would like to keep the Work In Progress section around for a little bit.

i.e. those are addins that I have "found" on GitHub, that haven't made it to NuGet.org yet.  I think that information is still useful, but perhaps, I should move it somewhere else.  Or perhaps, that information needs to make it over to the RFP and RFM issue repository.
